src/main/webapp/calendar-script.js

```diff
@@ -43,8 +66,125 @@ function getEvents(calendar) {
         title: event.name,
         start: event.strStartTime,
         end: event.strEndTime,
-        allDay: false
+        allDay: false,
+        extendedProps: {
+          stringDate: event.strStartTime,
+          address: event.address,
+          placeId: event.placeId, 
+          openTime: '9AM',
+          closeTime: '5PM',
+        }
       });
     });
   });
 }
+
+function createMap(modalBody, eventObj) {
+  const infoDisplay = document.createElement('div');
+  //infoDisplay.innerHTML = '<b>Address: </b><a href=' + url + 'target="_blank" >' + eventObj.extendedProps.address + '</a>' + '<br>';
+
+  const addressLine = document.createElement('p');
+  const boldAddress = document.createElement('b');
+  boldAddress.innerText = 'Address: ';
+  addressLine.appendChild(boldAddress);
+
+  const urlFormatAddress = eventObj.extendedProps.address.replace(/\s/g, '+')
+  const url = 'https://www.google.com/maps/search/?api=1&query=' + urlFormatAddress + '&query_place_id=' + eventObj.extendedProps.placeId;
+  const link = document.createElement('a');
+  link.href = url;
+  link.target = '_blank';
+  link.innerText = eventObj.extendedProps.address
+
+  addressLine.appendChild(link);
+  addressLine.appendChild(document.createElement('br'));
+  infoDisplay.appendChild(addressLine);
+
+  // infoDisplay.appendChild(address);
+  modalBody.appendChild(infoDisplay);
+
+  // create new div to hold map
+  const mapDis = document.createElement('div');
+  mapDis.id = 'map';
+  modalBody.appendChild(mapDis);
+
+  // instantiate map
+  const map = new google.maps.Map(document.getElementById('map'), {
+    zoom: zoomThirteen
+  });
+
+  const service = new google.maps.places.PlacesService(map);
+  service.getDetails({
+    placeId: eventObj.extendedProps.placeId
+  }, function(result, status) {
+    if (status != google.maps.places.PlacesServiceStatus.OK) {
+      alert(status);
+      return;
+    }
+    // set center of map
+    map.setCenter(result.geometry.location);
+  
+    // add open hours to mod info Display
+    let openHours = document.createElement('p');
+
+    // if there are open hours display them, otherwise open all day
+    try {
```

Comment:
only `openTime` and `closeTime` will throw an exception but if I create the HTML objects outside the try then its possible that it may display ```Open All Day``` ```Open: undefined, Close:undefinied```